glusterd2/xlator/options/options.go

```diff
@@ -236,7 +242,17 @@ func ValidatePriorityOrSize(o *Option, val string) error {
 
 // ValidateSizeList validates if the option is a valid size list
 func ValidateSizeList(o *Option, val string) error {
-	return ErrInvalidArg
+	l := len(val)
+	if strings.ContainsRune(val, 'B') {
+		t := strings.TrimSpace(val[:l-2])
+		v, err := strconv.ParseInt(t, 10, 64)
+		if err != nil {
+			return err
+		} else if v%512 != 0 {
```

Comment:
This isn't right. Where's this coming from ?

Reply:
I did the size validations here, intended to validate if element was a multiple of 512.
Missed the part that 'val' can have multiple values.
@prashanthpai ^^